### PR TITLE
feat(queue): wait for check that are not check-XXX

### PR DESCRIPTION
**Pull request by sileht**
To unqueue a pull request it must unmatch the pull_request_rules or
checks conditions in queue_rules. Otherwise we just wait the conditions
to become true.

Fixes MRGFY-496
Fixes MRGFY-491

### CODE REVIEW

File: mergify_engine/actions/merge_base.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ def strict_merge_parameter(v):
 async def get_rule_checks_status(
     ctxt: context.Context,
     rule: typing.Union["rules.EvaluatedRule", "rules.EvaluatedQueueRule"],
+    *,
+    unrelated_as_failure: bool = True,
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:106]: This name is not obvious to understand. Is there a better name or some docstring to help?

Comment (sileht) [reply mergify_engine/actions/merge_base.py]: I changed it

File: mergify_engine/actions/merge_base.py [hunk 2]
```diff
@@ -102,6 +102,8 @@ def strict_merge_parameter(v):
 async def get_rule_checks_status(
     ctxt: context.Context,
     rule: typing.Union["rules.EvaluatedRule", "rules.EvaluatedQueueRule"],
+    *,
+    other_unmatch_conditions_return_failure: bool = True,
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:106]:
```suggestion
    unmatched_conditions_return_failure: bool = True,
```
"other" does not mean much in this context 😢

File: mergify_engine/actions/merge_base.py [hunk 3]
```diff
@@ -102,6 +102,8 @@ def strict_merge_parameter(v):
 async def get_rule_checks_status(
     ctxt: context.Context,
     rule: typing.Union["rules.EvaluatedRule", "rules.EvaluatedQueueRule"],
+    *,
+    unmatch_conditions_return_failure: bool = True,
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:106]:
```suggestion
    unmatched_conditions_return_failure: bool = True,
```

**Review by jd — CHANGES_REQUESTED** [review]
You'll get this!!

File: mergify_engine/actions/merge_base.py (hunk 3 above)
Comment (sileht) [reply mergify_engine/actions/merge_base.py:106]: omg!!!!!

**Comment by sileht** [comment]
```
$ history -10000 | grep sed | tail -3
 9974  sed -i -e 's/unrelated_as_failure/other_unmatch_conditions_return_failure/g' mergify_engine/**/*.py
 9985  sed -i -e 's/other_unmatch_conditions_return_failure/unmatch_conditions_return_failure/g' mergify_engine/**/*.py
 9996  sed -i -e 's/unmatch_conditions_return_failure/unmatched_conditions_return_failure/g' mergify_engine/**/*.py
```

**Review by jd — APPROVED** [review]

**Pull request merged**